src/pages/Map.js

```diff
@@ -27,6 +38,44 @@ export default function Map() {
     right: 10,
     top: 10,
   };
+
+  useEffect(() => {
+    const url = "/api/pins.json";
+    fetch(url)
+      .then((res) => res.json())
+      .then((data) => {
+        setPins(data);
+      })
+      .catch((error) => {
+        console.log(error);
+      });
+  }, []);
+
+  const handleMarkerClick = (id, lat, long) => {
+    setCurrentPlaceId(id);
+    setViewport({ ...viewport, latitude: lat, longitude: long });
+  };
+
+  const handleAddClick = (event) => {
+    const [long, lat] = event.lngLat;
+    setNewPin({
+      lat,
+      long,
+    });
+  };
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    // eslint-disable-next-line
+    const newPlace = {
+      location,
+      date,
+      rating,
+      notes,
+      lat: newPin.lat,
+      long: newPin.long,
+    };
+  };
+
   return (
     <div className="Map">
       <ReactMapGl
```

Comment:
You should probably disable double click zoom if what you want to do when you do double click is set up a new marker:
https://visgl.github.io/react-map-gl/docs/api-reference/interactive-map#doubleclickzoom